### ISSUE TITLE
unnecessary qualification error on impl async fn

### ISSUE THREAD

**Issue by beckend** [rust-lang/rust-analyzer#16838]
**rust-analyzer version**:  2024-03-11 in releases page, not using VSCode.

**rustc version**:  1.78.0-nightly (3cbb93223 2024-03-13)

**relevant settings**:  unnecessary qualification
**code snippet to reproduce**: 
```rust
use async_trait;

#[async_trait::async_trait]
pub trait DropAsync {
  /// The method needed to drop.
  async fn async_drop(&mut self);
}

#[derive(Default)]
struct TestDropper<'a> {
  callback: Option<Box<dyn FnOnce() + Send + 'a>>,
}

#[async_trait::async_trait]
impl DropAsync for TestDropper<'static>
where
  Self: Send,
{
  async fn async_drop(&mut self) {
    todo!();
  }
}
```

impl error:

```shell
error: unnecessary qualification
   |
   |   async fn async_drop(&mut self) {
   |   ^^^^^
```

**Comment by flodiebold** [comment]
That's a rustc error, so you need to report this either at async_trait or rust-lang/rust.
